packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -0,0 +1,160 @@
+import { IAction } from './action';
+import { CfnPipeline } from './codepipeline.generated';
+
+/**
+ * Git push filter for trigger.
+ */
+export interface GitPushFilter {
+  /**
+   * The list of patterns of Git tags that, when pushed, are to be excluded from
+   * starting the pipeline.
+   *
+   * You can filter with glob patterns. The `excludedTags` takes priority
+   * over the `includedTags`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no tags.
+   */
+  readonly excludedTags?: string[];
+
+  /**
+   * The list of patterns of Git tags that, when pushed, are to be included as
+   * criteria that starts the pipeline.
+   *
+   * You can filter with glob patterns. The `excludedTags` takes priority
+   * over the `includedTags`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no tags.
+   */
+  readonly includedTags?: string[];
```

Comment:
In the original CFn, [GitPushFilter](https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-properties-codepipeline-pipeline-gitpushfilter.html) has `Tags`, and `Tags` has `Excludes` and `Includes`. However, the nesting is erased along [DESIGN_GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#flat).

I originally thought this was good, but today the trigger has [new feature](https://aws.amazon.com/jp/blogs/devops/aws-codepipeline-adds-support-for-branch-based-development-and-monorepos/) was added. It allows users to trigger branches as well as tags.

This feature has not yet been introduced to CFn, but perhaps like `Tags`, a property called `Branches` will be added to the `GitPushFilter`, which will also have `Excludes` and `Includes`. Considering this, I also thought it might be better to dare to nest `GitPushFilter` and have `GitTagFilterCriteria`.

Otherwise, the code would probably look something like this. (I am not even sure if this is a bad idea. I am thinking that it would be better to at least have a common prefix for each of them.)

```ts
export interface GitPushFilter {
  readonly excludedTags?: string[];
  readonly includedTags?: string[];
  readonly excludedBranches?: string[];
  readonly includedBranches?: string[];
  readonly excludedFilePaths?: string[];
  readonly includedFilePaths?: string[];
```

Reply:
I changed the property names for now: [change names to tagsExcludes and tagsIncludes](https://github.com/aws/aws-cdk/pull/28538/commits/95d17124c91cea2d2d7a1fef60a8abe6b13b4a9c)

If `Branches` is added as is, it would look like this:

```ts
export interface GitPushFilter {
  readonly tagsExcludes?: string[];
  readonly tagsIncludes?: string[];
  readonly branchesExcludes?: string[];
  readonly branchesIncludes?: string[];
  readonly filePathsExcludes?: string[];
  readonly filePathsIncludes?: string[];
```

Reply:
I like having what you've proposed.